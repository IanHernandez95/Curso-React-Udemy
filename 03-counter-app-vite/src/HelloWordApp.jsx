// export function App() {
//     // document.createElement....
//     return <h1>Hola Mundo</h1>;
// }

export const HelloWordApp = () => {
    return (
    <h1>Hello Word </h1>
    )
}
