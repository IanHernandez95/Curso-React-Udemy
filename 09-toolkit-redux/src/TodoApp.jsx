import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/apis/todosApi"


export const TodoApp = () => {

    const [todoID, setTodoID] = useState(1)

    // const { data: todos = [], isLoading } = useGetTodosQuery();
    const { data: todo, isLoading } = useGetTodoQuery( todoID );

    const nextTodo = () =>{
        setTodoID( todoID + 1);
    }

    const prevTodo = () => {
        if ( todoID === 1) return;
        setTodoID( todoID - 1);
    }

    return (
        <>
            <h1>Todos - RTK Query</h1>
            <hr />
            <h4>isLoading: {isLoading ? 'True' : 'False'}</h4>

            <pre>{JSON.stringify( todo )}</pre>


            <button onClick={ prevTodo }>
                prev todo
            </button>
            <button onClick={ nextTodo }>
                next todo
            </button>

            {/* <ul>
                { todos.map( todo => (
                    <li key={ todo.id }>
                        <strong>{ todo.completed ? 'Done: ' : 'Pending: ' }</strong>{ todo.title}
                    </li>
                ) ) }
            </ul> */}


        
        </>
    )
}
