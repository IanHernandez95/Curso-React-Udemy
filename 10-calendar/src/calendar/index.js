
export * from './pages/CalendarPage'

export * from './component/NavBar'
export * from './component/CalendarEvent'
export * from './component/CalendarModal'
export * from './component/FabAddNew'
export * from './component/FabDelete'