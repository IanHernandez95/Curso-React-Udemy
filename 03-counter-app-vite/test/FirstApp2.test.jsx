import { getAllByText, render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp', () => { 

    const title = 'Hola, soy un titulo'
    const subTitle = 'soy un subtitulo'

    test('Debe hacer Match con el Snapshot', () => {

        const { container } = render( <FirstApp title={ title }/>);
        expect( container ).toMatchSnapshot();

    });

    test('debe mostrar el mensaje "Hola, soy un titulo" ', () => {
        
        render( <FirstApp title={ title }/>);
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();

    });

    test('debe mostrar el titulo en un h1', () => {
        
        render( <FirstApp title={ title }/>);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML).toContain( title )

    });

    test('debe mostrar el subtitulo enviado por props', () => {
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />);

        expect( screen.getAllByText(subTitle).length).toBe(2);
    });

});