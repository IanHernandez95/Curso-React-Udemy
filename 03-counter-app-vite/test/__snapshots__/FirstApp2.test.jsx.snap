// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy un titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Ian Hernandez
  </p>
</div>
`;
